clients/src/main/java/org/apache/kafka/common/requests/MetadataRequest.java

```diff
@@ -25,24 +25,42 @@
 import java.util.List;
 
 public class MetadataRequest extends AbstractRequest {
-    
+
     private static final Schema CURRENT_SCHEMA = ProtoUtils.currentRequestSchema(ApiKeys.METADATA.id);
     private static final String TOPICS_KEY_NAME = "topics";
 
+    private final boolean allTopics;
     private final List<String> topics;
 
     public MetadataRequest(List<String> topics) {
+        this(topics, false);
+    }
+
+    /**
+     * If allTopics is true, then the passed topics will be ignored.
+     */
+    public MetadataRequest(List<String> topics, boolean allTopics) {
```

Comment:
@SinghAsDev The proposal is to use null on the wire protocol not in the API. In the discussions it was mentioned that being more explicit in the API was favored. 

Beyond that, if you specify `MetadataRequest(null)` java wont know which constructor to use between `MetadataRequest(List<String> topics)` and `MetadataRequest(Struct struct)`.